lxd/storage/backend_lxd.go

```diff
@@ -0,0 +1,353 @@
+}
+
+func (b *lxdBackend) SetInstanceQuota(i Instance, quota uint64) error {
+	return ErrNotImplemented
+}
+
+func (b *lxdBackend) MountInstance(i Instance) (bool, error) {
+	return true, ErrNotImplemented
+}
+
+func (b *lxdBackend) UnmountInstance(i Instance) (bool, error) {
+	return true, ErrNotImplemented
+}
+
+func (b *lxdBackend) GetInstanceDisk(i Instance) (string, string, error) {
+	return "", "", ErrNotImplemented
+}
+
+func (b *lxdBackend) CreateInstanceSnapshot(i Instance, name string, op *operations.Operation) error {
+	return ErrNotImplemented
+}
+
+func (b *lxdBackend) RenameInstanceSnapshot(i Instance, newName string, op *operations.Operation) error {
+	return ErrNotImplemented
+}
+
+func (b *lxdBackend) DeleteInstanceSnapshot(i Instance, op *operations.Operation) error {
+	return ErrNotImplemented
+}
+
+func (b *lxdBackend) RestoreInstanceSnapshot(i Instance, op *operations.Operation) error {
+	return ErrNotImplemented
+}
+
+func (b *lxdBackend) MountInstanceSnapshot(i Instance) (bool, error) {
+	return true, ErrNotImplemented
+}
+
+func (b *lxdBackend) UnmountInstanceSnapshot(i Instance) (bool, error) {
+	return true, ErrNotImplemented
+}
+
+func (b *lxdBackend) CreateImage(img api.Image, op *operations.Operation) error {
+	return ErrNotImplemented
+}
+
+func (b *lxdBackend) DeleteImage(img api.Image, op *operations.Operation) error {
+	return ErrNotImplemented
+}
+
+func (b *lxdBackend) CreateCustomVolume(vol api.StorageVolumesPost, op *operations.Operation) error {
+	project := "default"
+
+	volType, err := VolumeTypeNameToType(vol.Type)
+	if err != nil {
+		return err
+	}
+
+	// Create database entry for new storage volume.
+	err = VolumeDBCreate(b.state, b.name, vol.Name, vol.Description, vol.Type, false, vol.Config)
+	if err != nil {
+		return err
+	}
+
+	revert := true
+	defer func() {
+		if revert {
+			b.state.Cluster.StoragePoolVolumeDelete(project, vol.Name, volType, b.id)
+		}
+	}()
+
+	volID, _, err := b.state.Cluster.StoragePoolNodeVolumeGetTypeByProject(project, vol.Name, volType, b.id)
+	if err != nil {
+		return err
+	}
+
+	err = b.driver.CreateVolume(drivers.VolumeTypeCustom, volID, vol.Name, op)
+	if err != nil {
+		return err
+	}
+
+	revert = false
+	return nil
+}
+
+func (b *lxdBackend) CreateCustomVolumeFromCopy(vol api.StorageVolumesPost, op *operations.Operation) error {
+	project := "default"
+	volType, err := VolumeTypeNameToType(vol.Type)
+	if err != nil {
+		return err
+	}
+
+	// Check the supplied config and remove any fields not relevant for destination pool type.
+	volConfig, err := VolumePropertiesTranslate(vol.Config, b.driver.Name())
+	if err != nil {
+		return err
+	}
+
+	// Create database entry for new storage volume.
+	err = VolumeDBCreate(b.state, b.name, vol.Name, vol.Description, vol.Type, false, volConfig)
+	if err != nil {
+		return err
+	}
+
+	// Get the source pool's ID for use later copying snapshots.
+	srcPoolID, err := b.state.Cluster.StoragePoolGetID(vol.Source.Pool)
```

Comment:
Feels like we could probably stuff that in the struct at loading/creation time, would save us from having to query it, potentially multiple times down the line.

Reply:
@stgraber yes agreed, I'd also like to look into whether we can get the DB row create function to return the auto inc ID rather than having to query for it again.

Reply:
@stgraber this has been added.